### PR TITLE
Prune RDS backups on secondary account

### PR DESCRIPTION
**Pull request by ewjordan**
This is an AWS Lambda script that will run as a daily cron task on the backup AWS account. It prunes all but the latest 5 backups in the account that match the following conditions:
- contain 'production' in the name
- do not contain 'retain' in the name
- are manual backups (as opposed to automated)
- are in the 'available' state

As mentioned in the file, this is not automatically deployed, since we do not want anyone to have readily available credentials to run code on the backup account. I'll put together a summary of how to set up the backup account and get it on Confluence once I actually go through the process on the real backup account (currently I'm using my own AWS account).

Tests are coming shortly in another PR, I've tested each of the bits manually in the REPL and this is already running on my personal account, but I'm setting this up to be testable from the Lambda interface by passing a test parameter. This won't be automated, but will be useful to invoke manually before deploying changes to the function.

### CODE REVIEW

**Comment by ewjordan** [comment]
@wjordan @ashercodeorg @jeremydstone

**Comment by ewjordan** [comment]
Test failure is the "ActiveRecord::RecordNotFound: Level: {"name":"CSPU5_U3L13 Student Lesson Introduction","concept_ids":null}, Script: csp5" issue mentioned on Slack

**Comment by ewjordan** [comment]
I lied, I pushed tests into this PR instead of doing another

File: tools/scripts/prune-rds-backups.js [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+    // Delete each of the snapshots in the list
+    var deleteSnapshots = function (snapshots) {
+        var snapshotCount = snapshots.length;
+        for (var i=0; i<snapshotCount; ++i) {
+            var params = {
+                DBSnapshotIdentifier: snapshots[i].DBSnapshotIdentifier
+            };
+            console.log('Deleting snapshot with ID ' + snapshots[i].DBSnapshotIdentifier);
+            rds.deleteDBSnapshot(params, function (err, data) {
```
Comment (jeremydstone) [inline tools/scripts/prune-rds-backups.js:36]: Does rds.deleteDBSnapshot delete everything? I know when you delete an EC2 snapshot (although this is a different animal) through the UI, you have to delete the snapshot and then separately delete the EBS volume. I don't think there's the notion of EBS volumes that you manage separately with RDS, but thought I'd ask...

Comment (ewjordan) [reply tools/scripts/prune-rds-backups.js:36]: In the case of RDS, it's all just the snapshot (I think the closest equivalent of an inactive EBS volume would be a database in the stopped state), so it looks like this should do the trick. We generally won't be creating DBs out of these snapshots, so no worries there.

**Comment by ewjordan** [comment]
Just a little bump on this - it's been running on my AWS account for a while now, and has been keeping the count at 5, as expected. I'm holding off on setting this live on the real secondary account until I get a thumbs up.

FWIW, I looked into costs a little bit, and it looks like we're actually only charged for the actual compressed size of the database snapshot, which is often much smaller than the "face value" for 2TB/snapshot/month - so we might be able to get away with keeping more than 5 of these without breaking the bank. I haven't noticed _any_ increase in my personal AWS bill this month despite having 5 snaps stored for over a week, so it's possible that because these snapshots are connected to an active RDS database, they're not charging anything at all, in which case we can keep a lot more.

**Pull request merged**